### PR TITLE
SPORTS-62 Add APIs to create/edit/delete/fetch categories

### PR DESCRIPTION
**Pull request by sanjeevkpandit**
- `GET /categories`
- `POST /categories`
- `GET /categories/:id`
- `PUT /categories/:id`
- `DELETE /categories/:id`

### CODE REVIEW

File: backend/src/constants/messages.ts [hunk 1]
```diff
@@ -41,3 +41,15 @@ export const tokenMessages = {
...
+  notFound: 'Category not found.',
+  updated: 'Category updated successfully.',
+  removed: 'Category deleted successfully.',
+  fetched: 'Category(s) fetched successfully.',
+  created: 'New category created successfully.',
+  unableToRemove: 'Unable to remove the category.',
+  unableToUpdate: 'Unable to update the category.',
+  unableToCreate: 'Unable to create a new category.'
+}
+
```
Comment (aviskarkc10) [inline backend/src/constants/messages.ts]: are there two blank spaces here?

File: backend/src/constants/messages.ts [hunk 2]
```diff
@@ -41,3 +41,15 @@ export const tokenMessages = {
...
+export const categoryMessages = {
+  notFound: 'Category not found.',
+  updated: 'Category updated successfully.',
+  removed: 'Category deleted successfully.',
+  fetched: 'Category(s) fetched successfully.',
+  created: 'New category created successfully.',
+  unableToRemove: 'Unable to remove the category.',
+  unableToUpdate: 'Unable to update the category.',
+  unableToCreate: 'Unable to create a new category.'
+}
```
Comment (arun-tamang) [inline backend/src/constants/messages.ts]: semi-colon missing here.

File: backend/src/controllers/categoryController.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    const response = await CategoryService.getAll();
+
+    res.status(response.code).json(response);
+  } catch(error) {
+    next(error);
+  }
+}
+
+/**
+ * Get a category details.
```
Comment (aviskarkc10) [inline backend/src/controllers/categoryController.ts]: detail

File: backend/src/controllers/categoryController.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    const response = await CategoryService.get(req.params.id);
+
+    res.status(response.code).json(response);
+  } catch (error) {
+    next(error);
+  }
+}
+
+/**
+ * Update a category details.
```
Comment (aviskarkc10) [inline backend/src/controllers/categoryController.ts]: detail

File: backend/src/services/categoryService.ts [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+/**
+ * Create a new category.
+ *
+ * @export
+ * @param {object} params
+ * @returns {object}
+ * @throws {NoRowUpdatedError|error}
+ */
+export async function create(params: object) {
+  try {
```
Comment (aviskarkc10) [inline backend/src/services/categoryService.ts]: why are we nesting try catch?

Comment (Ishanliv99) [inline backend/src/services/categoryService.ts]: why nested try catch?

File: backend/src/validators/categoryValidator.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+ * @param res
+ * @param next
+ */
+export function validateCategorySchema(req: Request, res: Response, next: NextFunction) {
+  const result = validate(req.body, categorySchemas.categorySchema);
+
+  if (result.error) {
+    res.status(HttpStatus.BAD_REQUEST).json(result.error);
+
+    return;
```
Comment (aviskarkc10) [inline backend/src/validators/categoryValidator.ts]: do we need this return?

File: backend/src/services/categoryService.ts [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    return {
+      code: HttpStatus.OK,
+      data: updatedCategory,
+      message: categoryMessages.updated,
+      status: HttpStatus.getStatusText(HttpStatus.OK)
+    };
+  } catch (error) {
+    throw error;
+  }
+};
```
Comment (arun-tamang) [inline backend/src/services/categoryService.ts]: keep semi-colon after function consistent.

File: backend/src/validators/categoryValidator.ts (hunk 6 above)
Comment (sanjeevkpandit) [reply backend/src/validators/categoryValidator.ts]: Need `return` because the request moves on even using `res`.

**Pull request merged**